src/main/java/org/alfresco/opencmis/CMISConnector.java

```diff
@@ -3097,13 +3097,13 @@ public void setProperties(NodeRef nodeRef, TypeDefinitionWrapper type, Propertie
                     throw new CmisInvalidArgumentException("Property " + property.getId() + " is unknown!");
                 }
             }
+
```

Comment:
There should be a single check to see if the node is a working:
`boolean isOnWorkingCopy = checkOutCheckInService.isWorkingCopy(nodeRef);`

Reply:
The aim is to only check for the working copy once for all properties, reducing a whole bunch of unnecessary work back and forth to check aspects, etc.